tests/parallel_testsuite.py

```diff
@@ -48,8 +50,14 @@ def __init__(self, max_cores):
 
   def run(self, result):
     test_queue = self.create_test_queue()
-    self.init_processes(test_queue)
-    results = self.collect_results()
+    try:
+      self.init_processes(test_queue)
+      results = self.collect_results()
+    finally:
+      for p in self.processes:
+        p.terminate()
+        p.join()
+      self.processes.clear()
```

Comment:
what does `.clear()` do that is different from `self.processes = []`? (is it not a normal array somehow?)

Reply:
I assume the latter allocates a new array object/id?
Which could potentially cause problems if other objects are holding references to self.processes specifically.

If so: that's comment-worthy, cause it's a nonlocal concern.